src/react-hooks-nesting-walker/react-hooks-nesting-walker.ts

```diff
@@ -180,6 +180,21 @@ export class ReactHooksNestingWalker extends RuleWalker {
         return;
       }
 
+      /**
+       * Detect if the unnamed expression is wrapped in a illegal function call
+       */
+      if (
+        isCallExpression(ancestor.parent) &&
+        isIdentifier(ancestor.parent.expression) &&
```

Comment:
I'm wondering if this extra `isIdentifier` check is meaningful. I'm not saying this is wrong by any means. I'm just wondering if this case is different from the `if` below, where we check `isCallExpression` and `isReactComponentDecorator`, but not `isIdentifier` :thinking:

Reply:
It is. In fact, it is the only meaningful check that we need to do, given that we've reached that expression. All the others have been ruled out and returned early.